scripts/leapfrog/README.md

```diff
@@ -0,0 +1,136 @@
+# Leapfrog Documentation
+
+#### Overview
+A Leapfrog upgrade is a major upgrade that skips at least one release. Currently
+RPCO supports leapfrog upgrades from kilo to newton-14.1.
+
+#### Terms
+
+* [RPCO](https://github.com/rcbops/rpc-openstack): Rackspace Private Cloud powered by OpenStack
+* [OSA](https://github.com/openstack/openstack-ansible):  OpenStack Ansible
+* [OSA-OPS](https://github.com/openstack/openstack-ansible-ops):  OpenStack Operations
+* [Kilo](https://github.com/rcbops/rpc-openstack/tree/kilo): The RPCO release of OpenStack Kilo
+* [Newton-14.1](https://github.com/rcbops/rpc-openstack/tree/newton-14.1) The second RPCO release of OpenStack Newton. This version includes Leapfrog upgrade tools.
+
+## Executing a leapfrog upgrade
+
+The start point is a kilo deployment. It is useful to check the kilo deployment
+is healthy before starting the leapfrog process, for example by running tempest.
+
+```
+root@kilo:/opt/rpc-openstack/rpcd/playbooks# openstack-ansible ../../scripts/run_tempest.yml -t tempest_execute_tests -vv
+```
+
+If tempest raises issues, those should be fixed before continuing.
+
+The next step is to checkout the newton-14.1 branch of rpco. This will remove
+any uncommitted information from the rpco repo, so back that up if it is
+required.
+
+```
+root@kilo:/opt/rpc-openstack# git reset --hard
+git clean -df
+pushd openstack-ansible
+git reset --hard
+git clean -df
+popd
+git fetch origin
+git checkout newton-14.1
+git submodule update
+git status
+```
+
+It is crucial that the rpco repo and openstack-ansible submodule are both clean
+and at the appropriate SHAs for newton-14.1 at this stage. The [github
+newton-14.1 branch
+page](https://github.com/rcbops/rpc-openstack/tree/newton-14.1) shows both SHAs,
+the RPCO SHA is the latest commit, and the OSA SHA is shown as an annotation
+after @ on the openstack-ansible submodule.
+
+The next step is to execute the leapfrog upgrade script
+
```

Comment:
We do not need to go into the details of database maintenance. All we need to do is say that it should be done. We should definitely add it as a reminder above where the health check reminder is.